Marlin/src/sd/cardreader.cpp

```diff
@@ -472,50 +472,61 @@ void CardReader::mount() {
 #endif
 
 void CardReader::manage_media() {
-  static uint8_t prev_stat = 2;       // First call, no prior state
-  uint8_t stat = uint8_t(IS_SD_INSERTED());
-  if (stat == prev_stat) return;
+  //DEBUG_SECTION(cmm, "CardReader::manage_media()", true);
 
-  DEBUG_ECHOLNPGM("SD: Status changed from ", prev_stat, " to ", stat);
+  // First bit :media inserted , second bit : media initialized
+  static uint8_t prev_media_stat = 0;
 
-  flag.workDirIsRoot = true;          // Return to root on mount/release
+  uint8_t media_is_inserted = uint8_t(IS_SD_INSERTED());
+  if (media_is_inserted == (prev_media_stat & 1)) return; // if no insert or eject then return.
```

Comment:
We don't want to simply `return` in the case where there is no media during first-time initialization (from `setup()`). You will find that if you have no SD card inserted at boot, but then you insert one with a PLR file on it later, Power-Loss Recovery will be initiated. If you insert an SD card with an EEPROM file on it (and `SDCARD_EEPROM_EMULATION` enabled) then the printer will load the settings from EEPROM — even if the printer is in the middle of host printing. That is why the old code initializes `prev_stat` equal to 2. This method needs to initialize some things even when there is no media present.

Reply:
It looks like the call to `settings.first_load` in `setup()` will prevent the EEPROM from getting loaded later.